### PR TITLE
Add example for azure function app using key vault reference

### PR DESCRIPTION
**Pull request by xiaxyi**

### CODE REVIEW

**Comment by xiaxyi** [comment]
updated:

results:
```
No changes. Your infrastructure matches the configuration.
```

File: quickstart/201-function-app-key-vault-ref/main.tf [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  service_plan_id = azurerm_service_plan.default.id
+
+  storage_key_vault_secret_id = azurerm_key_vault_secret.default.id
+
+  identity {
+    type         = "UserAssigned"
+    identity_ids = [azurerm_user_assigned_identity.default.id]
+  }
+
+  key_vault_reference_identity_id = azurerm_user_assigned_identity.default.id
```
Comment (lonegunmanb) [inline quickstart/201-function-app-key-vault-ref/main.tf:96]: Could we adjust the arguments inside `resource` block by declaring arguments before nested blocks? Thanks.

File: quickstart/201-function-app-key-vault-ref/main.tf [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+  app_settings = {
```
Comment (lonegunmanb) [inline quickstart/201-function-app-key-vault-ref/main.tf:98]: Same here.

**Review by lonegunmanb — CHANGES_REQUESTED** [review]
Thanks @xiaxyi for opening this pr, almost LGTM except some tiny format issues.

File: quickstart/201-function-app-key-vault-ref/main.tf (hunk 1 above)
Comment (xiaxyi) [reply quickstart/201-function-app-key-vault-ref/main.tf:96]: Thanks for the comments @lonegunmanb , updated.

File: quickstart/201-function-app-key-vault-ref/main.tf (hunk 2 above)
Comment (xiaxyi) [reply quickstart/201-function-app-key-vault-ref/main.tf:98]: the same. Thanks

**Review by lonegunmanb — APPROVED** [review]
Thanks @xiaxyi , LGTM!

**Pull request merged**